pisa/population.py

```diff
@@ -0,0 +1,222 @@
+import urllib
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+
+import numpy as np
+import pandas as pd
+import requests
+import rasterio
+from rasterio.mask import mask
+from geopandas import GeoDataFrame, clip, points_from_xy
+from shapely import MultiPolygon, Polygon
+
+from hdx.api.configuration import Configuration
+from hdx.data.resource import Resource
+
+from pisa.administrative_area import AdministrativeArea
+
+
+@dataclass
+class DataSource(Enum):
+    WORLD_POP = 'world_pop'
+    FACEBOOK = 'facebook'
+
+
+class Population(ABC):
+    """Abstract base class for Population. Subclasses must implement the method get_population_data()
+    If you want to add new data source (e.g. geojson): create new subclass with method get_population_data().
+    """
+    def __init__(
+        self,
+        admin_area: AdministrativeArea,
+        admin_boundaries: Polygon | MultiPolygon,
```

Comment:
I played a bit with the notebook and I have to take back a previous comment: I thought that an `AdministrativeArea` object would already store both the admin boundaries and the country name, but I forgot that to get the boundaries the user must call the `get_admin_area_boundaries` method with the name of the admin area desired. So my previous suggestion of having the admin area object as an argument was because by passing it we wouldn't need to also pass the boundaries, but of course I was wrong. 

So if you would like to revert the change and have the country code as an argument instead of the boundaries, feel free to do it! Otherwise this also looks good :)